language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java

```diff
@@ -223,56 +229,40 @@ public void sort(BallerinaCompletionContext context, T node, List<LSCompletionIt
                     }
                     String label = prefix;
                     String insertText = prefix;
-                    CompletionItem item = new CompletionItem();
-                    item.setLabel(label);
-                    item.setInsertText(insertText);
-                    item.setDetail(ItemResolverConstants.MODULE_TYPE);
-                    item.setKind(CompletionItemKind.Module);
-                    populatedList.add(orgName + "/" + pkgName);
+                    CompletionItem item = this.getModuleCompletionItem(label, insertText, new ArrayList<>());
+                    processedList.add(processedModuleHash);
                     return new SymbolCompletionItem(ctx, null, item);
                 }).collect(Collectors.toList());
 
         List<Package> packages = LSPackageLoader.getInstance(ctx.languageServercontext()).getDistributionRepoPackages();
-        // TODO: Refactor to match the latest project structure
-//        packages.addAll(LSPackageLoader.getCurrentProjectModules(currentPkg, ctx));
         packages.forEach(pkg -> {
             String name = pkg.packageName().value();
             String orgName = pkg.packageOrg().value();
-            if (!CommonUtil.matchingImportedModule(ctx, pkg) && !populatedList.contains(orgName + "/" + name)
-                    && !this.isPreDeclaredLangLib(pkg)) {
-                CompletionItem item = new CompletionItem();
-                item.setLabel(CommonUtil.getPackageLabel(pkg));
+            if (!CommonUtil.matchingImportedModule(ctx, pkg) && !processedList.contains(orgName + "/" + name)) {
```

Comment:
Same as above. We can use CommonKeys#SLASH_KEYWORD_KEY for the "/" right?

Reply:
Fixed